File: bot/commands/admin/prefix.js
exports.run = async (bot) => {
  bot.registerCommand("prefix", async (message, args) => {
    bot.database.Guilddata.findOne({ guildID: message.channel.guild.id }, async (err, guilddata) => {
      if (err) bot.log("error", err)
      if (bot.checkBannedUsers(message.author.id)) return
      if (!await bot.checkBotPermission(message, ["readMessages", "sendMessages"])) return

      let prefix = args.join(" ").replace("{space}", " ")

      if (prefix.length > 20) return bot.createMessage(message.channel.id, {embed:{title:"Error",color:bot.color.red,description:"Prefix must be less then `20` characters long",timestamp:new Date()}})

      if (!args[0]) {
        const prefixEmbed = {
          embed: {
            title: `Prefix for ${message.channel.guild.name}`,
            color: bot.getEmbedColor(bot, message),
            description: `Current prefix: \`${guilddata.prefix == "" ? "a!" : guilddata.prefix}\``,
            timestamp: new Date()
          }
        }
        await bot.createMessage(message.channel.id, prefixEmbed)
      }

      if (prefix) {
        if (message.member.permission.has("administrator")) {
          bot.database.Guilddata.findOneAndUpdate({ guildID: message.channel.guild.id }, {$set: {"prefix":prefix}})
          bot.registerGuildPrefix(message.channel.guild.id, [prefix, "@mention"])
          const newPrefix = {
            embed: {
              title: `New prefix for ${message.channel.guild.name}`,
              color: bot.getEmbedColor(bot, message),
              description: `New prefix is: \`${prefix}\``,
              timestamp: new Date()
            }
          }
          bot.createMessage(message.channel.id, newPrefix)
        } else {
          return bot.createMessage(message.channel.id, {embed:{title:"Error",color:bot.color.red,description:"Only admins can change the server prefix",timestamp:new Date()}})
        }
      }

    })
  }, {
    cooldown: 3000,
    cooldownMessage: "Whoa there slow down, the cooldown is 3 seconds!"
  })
}

File: src/main/java/dastrangeboi/atlantis/Commands/Prefix.java
package dastrangeboi.atlantis.Commands;

import dastrangeboi.atlantis.Bot;
import dastrangeboi.atlantis.Database;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import org.bson.Document;
import java.awt.*;
import java.util.Objects;

public class Prefix {

    private static Color getRoleColor(Member member) {
        return member.getColor();
    }

    public static void run(GuildMessageReceivedEvent message, String[] args) {

        Document guildFound = Database.guilddata.find(new Document("guildID", message.getGuild().getId())).first();
        if (guildFound == null) return;

        String prefix = String.join(" ", args);

        if (prefix.length() > 20) {
            EmbedBuilder errorEmbed = new EmbedBuilder()
                    .setTitle("Error")
                    .setColor(Color.red)
                    .setDescription("Prefix must be less then `20` characters long");
            message.getChannel().sendMessage(errorEmbed.build()).queue();
        }

        if (args.length < 1) {
            EmbedBuilder prefixEmbed = new EmbedBuilder()
                    .setTitle("Prefix for " + message.getGuild().getName())
                    .setColor(getRoleColor(message.getGuild().getSelfMember()))
                    .setDescription("Current prefix: `" + (guildFound.getString("prefix").equalsIgnoreCase("") ? Bot.prefix : guildFound.getString("prefix")) + "`");
            message.getChannel().sendMessage(prefixEmbed.build()).queue();
        }

        if (prefix.length() > 0) {
            if (Objects.requireNonNull(message.getMember()).hasPermission(Permission.ADMINISTRATOR)) {
                Database.guilddata.updateOne(new Document("guildID", message.getGuild().getId()), new Document("$set", new Document("prefix", prefix)));
                EmbedBuilder newPrefixEmbed = new EmbedBuilder()
                        .setTitle("New prefix for " + message.getGuild().getName())
                        .setColor(getRoleColor(message.getGuild().getSelfMember()))
                        .setDescription("New prefix is: `" + prefix + "`");
                message.getChannel().sendMessage(newPrefixEmbed.build()).queue();
            } else {
                EmbedBuilder errorEmbed = new EmbedBuilder()
                        .setTitle("Error")
                        .setColor(Color.red)
                        .setDescription("Only admins can change the server prefix");
                message.getChannel().sendMessage(errorEmbed.build()).queue();
            }
        }
    }
}
